frontend/public/components/catalog-item-icon.tsx

```diff
@@ -195,3 +195,19 @@ export type ClusterServiceClassIconProps = {
   serviceClass: K8sResourceKind,
   iconSize?: string,
 };
+
+export const ImageStreamIcon: React.SFC<ImageStreamIconProps> = ({tag, iconSize}) => {
```

Comment:
@dtaylor113 FYI, this adds an image stream icon

Reply:
I had to break this into separate methods which return just the `iconClass` and/or `imgUrl` separate from the HTML elements because `patterfly-react-extension's CatalogTileView` does it's own HTML rendering.
https://github.com/openshift/console/pull/530/files#diff-9e67038ceaed601f6ac700bc2e55d7ddR180